File: 프로그래머스/2/49993. 스킬트리/스킬트리.cs
using System;

public class Solution {
    public int solution(string skill, string[] skill_trees) {
        int answer = 0;
        
        foreach(var tree in skill_trees)
        {
            int expected = 0;
            bool isOk = true;
            
            foreach(char c in tree)
            {
                int idx = skill.IndexOf(c);
                if(idx == -1) continue;
                if(idx == expected) expected++;
                else
                {
                    isOk = false; break;    
                }
            }
            if(isOk) answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/49993. 스킬트리/스킬트리.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer = 0;
    
    for(string tree : skill_trees)
    {
        int idx = 0;
        bool valid = true;
        
        for(char c : tree)
        {
            if(skill.find(c) == string::npos) continue;
            
            if(c == skill[idx])
            {
                idx++;
            }
            else
            {
                valid = false;
                break;
            }
        }
        if(valid) answer++;
    }
    return answer;
}